pulsar/consumer_regex.go

```diff
@@ -360,13 +235,13 @@ func (c *regexConsumer) unsubscribe(topics []string) {
 	}
 }
 
-func (c *regexConsumer) topics() ([]string, error) {
-	topics, err := c.client.lookupService.GetTopicsOfNamespace(c.namespace, internal.Persistent)
+func topics(client *client, namespace string, pattern *regexp.Regexp) ([]string, error) {
+	topics, err := client.lookupService.GetTopicsOfNamespace(namespace, internal.Persistent)
```

Comment:
It should check the `client` for `nil`.

Reply:
This is a private function, we ensure the client is not nil. If the client is nil, I think panic is acceptable.

Reply:
Can we make this function as part of struct method? So that client would not be nil. 
func (c *regexConsumer) topics() ..{
  topics, err: =c.client